### PR TITLE
Final changes to Thermodynamics.jl 

### PR DESCRIPTION
**Pull request by yairchn**
this PR should be revised once this PR in Thermodynamics.jl is merged 
https://github.com/CliMA/Thermodynamics.jl/pull/52#issuecomment-930411505

### CODE REVIEW

**Review by charleskawczynski — COMMENTED** [review]
Looks good to me! Let's see what buildkite says ( I pushed a commit for us to upgrade to the new version).

**Comment by yairchn** [comment]
Bors r+

**Review by charleskawczynski — APPROVED** [review]

**Comment by yairchn** [comment]
Bors r+

**Comment by yairchn** [comment]
> Bors r+

Hopefully the last one for this PR!

File: Project.toml [hunk 1]
```diff
@@ -36,7 +36,6 @@ UnPack = "3a884ed6-31ef-47d7-9d2a-63182c4928ed"
 ArgParse = "1.1"
 ArtifactWrappers = "0.1"
 CLIMAParameters = "0.2"
-ClimaCore = "0.2"
```
Comment (charleskawczynski) [inline Project.toml:39]: We need to keep this.

**Comment by yairchn** [comment]
Bors r+

File: integration_tests/utils/parameter_set.jl [hunk 2]
```diff
@@ -13,6 +13,7 @@ CLIMAParameters.Planet.cp_v(ps::EarthParameterSet) = ps.nt.cp_v
 CLIMAParameters.Planet.R_d(ps::EarthParameterSet) = ps.nt.R_d
 CLIMAParameters.Planet.R_v(ps::EarthParameterSet) = ps.nt.R_v
 CLIMAParameters.Planet.molmass_ratio(ps::EarthParameterSet) = ps.nt.molmass_ratio
+CLIMAParameters.Planet.T_freeze(ps::EarthParameterSet) = ps.nt.molmass_ratio
```
Comment (yairchn) [inline integration_tests/utils/parameter_set.jl:16]: @trontrytel mind this when you are considering to add ice to the model

File: integration_tests/utils/parameter_set.jl [hunk 3]
```diff
@@ -47,6 +48,7 @@ function create_parameter_set(namelist)
         cp_v = 1859.0,
         R_d = 287.1,
         R_v = 461.5,
+        T_freeze = 0.0, # TODO: no ice until it is properly added to the model
```
Comment (yairchn) [inline integration_tests/utils/parameter_set.jl:51]: @trontrytel mind this when you are considering to add ice to the model

File: integration_tests/utils/parameter_set.jl (hunk 2 above)
Comment (trontrytel) [reply integration_tests/utils/parameter_set.jl:16]: Why is the freezing temperature set to molar mass ratio?

Comment (trontrytel) [reply integration_tests/utils/parameter_set.jl:16]: @yairchn  - this is a bug, right?

File: integration_tests/utils/generate_namelist.jl [hunk 4]
```diff
@@ -256,7 +256,7 @@ function Rico(namelist_defaults)
     namelist["grid"]["dz"] = 50.0
 
     namelist["time_stepping"]["t_max"] = 86400.0
-    namelist["time_stepping"]["dt"] = 8.0
+    namelist["time_stepping"]["dt"] = 7.0
```
Comment (trontrytel) [inline integration_tests/utils/generate_namelist.jl:260]: Why do we have to reduce the timestep here?

File: integration_tests/utils/parameter_set.jl (hunk 3 above)
Comment (trontrytel) [reply integration_tests/utils/parameter_set.jl:51]: copy!

File: integration_tests/utils/parameter_set.jl (hunk 2 above)
Comment (charleskawczynski) [reply integration_tests/utils/parameter_set.jl:16]: Good catch!

**Review by trontrytel — APPROVED** [review]
found one bug. Looks great otherwise!

File: integration_tests/utils/parameter_set.jl (hunk 2 above)
Comment (yairchn) [reply integration_tests/utils/parameter_set.jl:16]: copy paste bug

**Comment by yairchn** [comment]
Bors r+

File: integration_tests/utils/parameter_set.jl (hunk 2 above)
Comment (charleskawczynski) [reply integration_tests/utils/parameter_set.jl:16]: @yairchn

**Comment by charleskawczynski** [comment]
Let's fix the T_freeze = molar mass bug

**Comment by yairchn** [comment]
> Let's fix the T_freeze = molar mass bug

sorry I was sure I did. now it ok

**Comment by yairchn** [comment]
Bors r+

File: integration_tests/utils/generate_namelist.jl (hunk 4 above)
Comment (charleskawczynski) [reply integration_tests/utils/generate_namelist.jl:260]: Not a blocker, but was this necessary @yairchn?

Comment (yairchn) [reply integration_tests/utils/generate_namelist.jl:260]: Bors r-

Comment (yairchn) [reply integration_tests/utils/generate_namelist.jl:260]: I will check again to make sure it is still needed

Comment (yairchn) [reply integration_tests/utils/generate_namelist.jl]: it works with 8sec. I think but was breaking before T_freeze was set to zero on rainfall speed tilmestep warning - I pushed -a change back to dt=8 great catch @trontrytel

**Comment by yairchn** [comment]
Bors r+

**Comment by yairchn** [comment]
Bors r+

**Comment by charleskawczynski** [comment]
Closes #144.

**Comment by charleskawczynski** [comment]
Hmm, looks like random failure.

**Pull request merged**